rust/headless-client/src/lib.rs

```diff
@@ -66,8 +67,11 @@ pub(crate) const GIT_VERSION: &str = git_version::git_version!(
 const TOKEN_ENV_KEY: &str = "FIREZONE_TOKEN";
 
 /// CLI args common to both the IPC service and the headless Client
-#[derive(clap::Args)]
+#[derive(clap::Parser)]
 struct CliCommon {
+    #[arg(long, env = "FIREZONE_DNS_CONTROL")]
```

Comment:
I think you can set a default here:

```suggestion
    #[arg(long, env = "FIREZONE_DNS_CONTROL", default_value = "systemd-resolved")]
```

Reply:
And get rid of the option.